ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -72,40 +102,78 @@ export default class ConfigureAwsComponent extends Component<Args> {
-    const rootAttrChanged =
-      Object.keys(rootConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
-
-    if (!leaseAttrChanged && !rootAttrChanged) {
-      this.flashMessages.info('No changes detected.');
-      this.transition();
-    }
+  save = task(
+    waitFor(async (event: Event) => {
+      event.preventDefault();
+      this.resetErrors();
+      const { leaseConfig, rootConfig } = this.args;
+      // Note: aws/root-config model does not have any validations
+      const isValid = this.validate(leaseConfig);
+      if (!isValid) return;
+
+      // do issuer save and side effects if enterprise and isser saved
+      if (this.version.isEnterprise && this.issuerConfig.dirty) {
+        await this.updateIssuer.perform(this.issuerConfig);
+      }
+
+      // Check if any of the models' attributes have changed.
+      // If no changes to either model, transition and notify user.
+      // If changes to either model, save the model(s) that changed and notify user.
+      // Note: "backend" dirties model state so explicity ignore it here.
 
-    const rootSaved = rootAttrChanged ? yield this.saveRoot() : false;
-    const leaseSaved = leaseAttrChanged ? yield this.saveLease() : false;
+      const leaseAttrChanged =
+        Object.keys(leaseConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
+      const rootAttrChanged =
+        Object.keys(rootConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
 
-    if (rootSaved || leaseSaved) {
-      this.transition();
-    } else {
-      // otherwise there was a failure and we should not transition and exit the function.
+      if (!leaseAttrChanged && !rootAttrChanged) {
+        this.flashMessages.info('No changes detected.');
+        this.transition();
+      }
+
+      const rootSaved = rootAttrChanged ? await this.saveRoot() : false;
+      const leaseSaved = leaseAttrChanged ? await this.saveLease() : false;
+
+      if (rootSaved || leaseSaved) {
+        this.transition();
+      } else {
+        // otherwise there was a failure and we should not transition and exit the function.
+        return;
+      }
+    })
+  );
+
+  updateIssuer = task(async (issuerConfig: IssuerConfig) => {
+    if (issuerConfig.dirty && issuerConfig.original) {
+      // show modal with warning that the config will change
+      this.saveIssuerWarning = 'Issuer will be updated. Continue?';
+      while (this.saveIssuerWarning) {
+        if (Ember.testing) {
+          return;
+        }
+        await timeout(500);
+      }
+    } else if (issuerConfig.dirty && issuerConfig.noRead) {
+      // show modal with warning that the config *may* update existing configs
+      this.saveIssuerWarning = 'Issuer may overwrite previous value. Continue?';
+      while (this.saveIssuerWarning) {
+        if (Ember.testing) {
+          return;
+        }
+        await timeout(500);
+      }
+    }
+    try {
+      return this.store
```

Comment:
If you're returning here, will the flash message still appear?

Reply:
nope -- draft mode :)